packages/frontend/src/components/common/RouterNavLink.tsx

```diff
@@ -2,27 +2,24 @@ import { NavLink, type NavLinkProps } from '@mantine/core';
 import { type FC } from 'react';
 import {
     NavLink as ReactRouterNavLink,
-    useRouteMatch,
+    useLocation,
+    useMatch,
     type NavLinkProps as ReactRouterNavLinkProps,
-} from 'react-router-dom';
+} from 'react-router-dom-v5-compat';
 
-type RouterNavLinkProps = Omit<NavLinkProps, 'component' | 'active'> &
-    Omit<ReactRouterNavLinkProps, 'component'>;
+type RouterNavLinkProps = Omit<NavLinkProps, 'component' | 'active'> & {
+    exact?: boolean;
+} & Omit<ReactRouterNavLinkProps, 'component'>;
 
 const RouterNavLink: FC<RouterNavLinkProps> = (props) => {
-    const match = useRouteMatch(props.to.toString());
-
+    const location = useLocation();
+    const exactMatch = useMatch(props.to.toString());
+    const isPartialMatch = location.pathname.startsWith('/example');
```

Comment:
i don't get this bit where we check if it starts with /example

Reply:
```suggestion
    const isPartialMatch = location.pathname.startsWith(props.to.toString());
```